akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -111,6 +112,18 @@ object Source {
   }
 
   /**
+   * Creates [[Source]] with `start` as the first element and each next element as `previous + 1` until
+   * it reaches `end`. It allows to create `Source` out of range as simply as on scala `Source(1 to N)`
+   */
+  def range(start: Int, end: Int): javadsl.Source[Integer, Unit] = {
+    require(start <= end, "start must be less or equal than end")
+    from(new util.AbstractList[Integer]() {
+      override def get(index: Int) = start + index
+      override def size() = end - start + 1
```

Comment:
Random thought... I seem to remember Scala's Range was "pretty heavy" though can't recall completely in what way...? So since `javadsl.Source.range(1, 10)` wants to be the equivalent of `scaladsl.Source(1 to 10)`, could we dispel my worry that if we don't provide a `range` in Scala it would perform worse? (I.e. do we not want to provide `range` for scala as well?)